provision/aws/aws_check_subnet_groups.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+if [[ "$RUNNER_DEBUG" == "1" ]]; then
+  set -x
+fi
+
+function check_db_subnet_group_usage() {
+    local db_subnet_group_name=$1
+    local region=$2
+    # Check for any RDS instances associated with the subnet group
+    rds_instances=$(aws rds describe-db-instances --region "$region" --query "DBInstances[?DBSubnetGroup.DBSubnetGroupName=='${db_subnet_group_name}'].DBInstanceIdentifier" --output text)
+    # Check for any Aurora clusters associated with the subnet group
+    rds_clusters=$(aws rds describe-db-clusters --region "$region" --query "DBClusters[?DBSubnetGroup=='${db_subnet_group_name}'].DBClusterIdentifier" --output text)
+    if [[ -z $rds_instances && -z $rds_clusters ]]; then
+       return 0 # No active resources
+    else
+       return 1 # Active resources found
+    fi
+}
+
+EXIT_CODE=0
+
+# Iterate through each DB subnet group and delete if not in use
+for region in $(aws account list-regions --query "Regions[?RegionOptStatus != `DISABLED`].RegionName" --output text); do
+    db_subnet_groups=$(aws rds describe-db-subnet-groups --region "$region" --query "DBSubnetGroups[].DBSubnetGroupName" --output text)
+    echo "Looking in AWS Region: $region"
+    for db_subnet_group in $db_subnet_groups; do
+        #echo "Checking DB Subnet Group: $db_subnet_group"
+        if check_db_subnet_group_usage "$db_subnet_group" "$region"; then
+            echo "Found an unused DB Subnet Group: $db_subnet_group in $region"
+            #we could use this script bit, to delete if necessary
+            #aws rds delete-db-subnet-group --db-subnet-group-name $db_subnet_group
```

Comment:
I would say it's best for us to no automatically delete the resources, as we want to determine where the resource leak originated and fix that.

Reply:
We can't remove them as these resources can originate from `keep-alive` deployments.

Reply:
that is another valid point as well. Thanks Michal, its removed from the script now.